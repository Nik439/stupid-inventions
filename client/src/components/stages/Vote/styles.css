.vote-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: flex-end;
  width: 100%;
  max-width: 800px;
  margin: 2rem auto;
}

.vote-left {
  margin: 0.5rem;
  align-items: center;
  height: 3.5rem;
  display: flex;
  font-weight: 600;
  font-size: 1.5em;
}

.vote-invention {
  width: 100%;
  max-width: 30rem;
  margin: 1rem;
}

.vote-title {
  width: 100%;
  text-align: center;
  hyphens: auto;
}

.vote-canvas {
  background-color: white;
  width: 100%;
  height: auto;
}

.vote-counter {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 3rem;
}

.vote-coin {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 2.5rem;
  width: 2.5rem;
  margin: 0.5rem;
  background-color: #ffd700;
  border: 2px solid #d9a225;
  border-radius: 1.5rem;
  box-shadow: inset 0 0 0 1px #00000044, 1px 1px 2px #00000044;
}
.vote-coin::before {
  content: '1';
  color: #d9a225;
  text-shadow: 0 0 0 1px #00000044;
  font-size: 1.5rem;
  font-weight: 600;
}
