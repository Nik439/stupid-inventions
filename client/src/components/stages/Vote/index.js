import React, {useState} from 'react';
import useTimer from '../../../hooks/timer';
import './styles.css';

function Vote(props) {
  const [votes, setVotes] = useState(1);

  const timer = useTimer(30000, () => props.doneVoting());

  function handleVote(e) {
    if (votes <= 3) {
      document.getElementById('vote-left').children[0].remove();
      e.target.parentNode.querySelector('.vote-counter').innerHTML +=
        "<span class='vote-coin'></span>";
      setVotes(votes + 1);
      props.submitVote(e.target.parentNode.querySelector('.vote-canvas').id);
      if (votes === 3) {
        props.doneVoting();
      }
    }
  }

  const drawingsList = props.drawingsList.map(drawing => {
    if (drawing.playerName !== props.userName) {
      return (
        <div
          key={drawing.playerName}
          className="vote-invention"
          onClick={e => handleVote(e)}
          data-testid="drawing"
        >
          <h3 className="vote-title">{drawing.name}</h3>
          <p style={{marginBottom: '1rem'}}>
            Click on the image below to vote for this drawing
          </p>
          <img
            id={drawing.playerName}
            className="vote-canvas"
            src={drawing.url}
            alt={`${drawing.playerName}'s drawing`}
          ></img>
          <div className="vote-counter"></div>
        </div>
      );
    } else return null;
  });

  return (
    <React.Fragment>
      <p className="vote-left" id="vote-left">
        Votes left: <span className="vote-coin" data-testid="coin"></span>
        <span className="vote-coin" data-testid="coin"></span>
        <span className="vote-coin" data-testid="coin"></span>
      </p>
      <p style={{marginLeft: '.5rem'}}>
        Time Remaining: {Math.ceil(timer / 1000)}
      </p>
      <div className="vote-container">{drawingsList}</div>
    </React.Fragment>
  );
}

export default Vote;
