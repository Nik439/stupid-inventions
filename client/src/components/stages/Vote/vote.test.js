import {fireEvent, render, screen} from '@testing-library/react';
import Vote from '.';

describe('Vote', () => {
  test('should render the correct amount of coins', () => {
    const drawing = {playerName: 'a', name: 'test', url: ''};

    render(
      <Vote
        drawingsList={[drawing]}
        submitVote={() => {}}
        doneVoting={() => {}}
        userName="b"
      />,
    );

    expect(screen.getAllByTestId('coin').length).toBe(3);
    fireEvent.click(screen.getByTestId('drawing'));
    expect(screen.getAllByTestId('coin').length).toBe(2);
    fireEvent.click(screen.getByTestId('drawing'));
    expect(screen.getAllByTestId('coin').length).toBe(1);
    fireEvent.click(screen.getByTestId('drawing'));
    expect(screen.queryByTestId('coin')).toBeNull();
  });

  test('should not display user image', () => {
    const playerName = 'a';
    const drawing = {playerName, name: 'test', url: ''};

    render(
      <Vote
        drawingsList={[drawing]}
        submitVote={() => {}}
        doneVoting={() => {}}
        userName="a"
      />,
    );

    expect(screen.queryByAltText(`${playerName}'s drawing`)).toBeNull();
  });

  test('should render oponent images', () => {
    const drawing = {playerName: 'b', name: 'test', url: ''};

    render(
      <Vote
        drawingsList={[drawing]}
        submitVote={() => {}}
        doneVoting={() => {}}
        userName="a"
      />,
    );

    expect(screen.queryAllByTestId('drawing')).toMatchSnapshot();
  });
});
