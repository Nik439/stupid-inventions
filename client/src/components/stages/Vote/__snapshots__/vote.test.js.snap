// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vote should render oponent images 1`] = `
Array [
  <div
    class="vote-invention"
    data-testid="drawing"
  >
    <h3
      class="vote-title"
    >
      test
    </h3>
    <img
      alt="a's drawing"
      class="vote-canvas"
      id="a"
      src=""
    />
    <div
      class="vote-counter"
    />
  </div>,
]
`;
